.github/workflows/cli.yml

```diff
@@ -42,52 +42,124 @@ concurrency:
+        run: yarn typecheck
+
+      - name: 🧪 E2E Test CLI
+        working-directory: packages/@expo/cli
+        run: yarn test:e2e --shard ${{ matrix.shard }}/${{ strategy.job-total }}
+
+  e2e-windows:
+    runs-on: windows-2022
+    continue-on-error: true
+    steps:
+      - name: 🪟 Setup Windows
+        shell: bash
+        # Configure Git to use LF line endlings, and Bun to use a faster cache disk on Windows
+        run: |
+          git config --global core.autocrlf false
+          git config --global core.eol lf
+          echo "BUN_INSTALL_CACHE_DIR=$GITHUB_WORKSPACE\.bun\install\cache" >> $GITHUB_ENV
+      
+      - name: 🏗️ Setup Node
+        uses: actions/setup-node@v4
+        with:
+          node-version: 18
+
       - name: 🏗️ Setup Bun
-        uses: oven-sh/setup-bun@v1
+        uses: oven-sh/setup-bun@v2
         with:
           # Version `1.x` fails due to https://github.com/oven-sh/setup-bun/issues/37
           # TODO(cedric): swap `latest` back once the issue is resolved
           bun-version: latest
 
-      - name: 🔎 Type Check CLI
-        run: yarn typecheck
-        working-directory: packages/@expo/cli
-      - name: E2E Test CLI
-        run: yarn test:e2e --shard ${{ matrix.shard }}/${{ strategy.job-total }}
+      - name: 👀 Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 100
+      - name: ⬇️ Fetch commits from base branch
+        run: git fetch origin ${{ github.event.before || github.base_ref || 'main' }}:${{ github.event.before || github.base_ref || 'main' }} --depth 100
+        if: github.event_name == 'pull_request' || github.event_name == 'workflow_dispatch'
+
+      - name: 🔎 Find Yarn Cache
+        id: yarn-cache
+        run: echo "dir=$(yarn cache dir)" >> $GITHUB_OUTPUT
+        shell: bash
+      - name: ♻️ Restore Yarn Cache
+        uses: actions/cache@v4
+        with:
+          path: ${{ steps.yarn-cache.outputs.dir }}
+          key: ${{ runner.os }}-yarn-${{ hashFiles('yarn.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-yarn-
+
+      - name: 🧶 Install node modules in root dir
+        # NOTE(cedric): yarn v1 on Windows has networking issues, we need to set `--network-timeout` to a higher value
+        run: yarn install --prefer-offline --frozen-lockfile --network-timeout 1000000
+        env:
+          YARN_IGNORE_SCRIPTS: 'true'
+
+      - name: 👷 Build Expo CLI
+        run: yarn workspace @expo/cli prepare
+
+      - name: 🔎 Type Check CLI (Windows)
         working-directory: packages/@expo/cli
+        # TODO(cedric): expo module scripts is incompatible with Windows, forcing us to manually set up these steps instead of `yarn typecheck`
+        run: yarn tsc --noEmit
 
+      - name: 🧪 E2E Test CLI (Windows)
+        working-directory: packages/@expo/cli
+        # NOTE(cedric): Windows is significantly slower and suffers from IO congestion when running more than 2 workers
+        run: yarn test:e2e --max-workers 2
```

Comment:
I've also embedded this into #33401, where it switches tests to run in serial mode when GHA is running in debug mode. Probably can drop the `--max-workers 2` here.

Reply:
Another requirement for this is that we have to modify the temporary system folder on Windows/GHA to be on the same disk as the repository checkout. The issue there is that Metro's FileMap only "scopes" a single disk on Windows [due to its design](https://github.com/facebook/metro/blob/main/packages/metro-file-map/src/lib/TreeFS.js#L96). Meaning that on Windows, it only scopes to `C:\\...` or `D:\\...`.

GHA uses both `C:` (system temp dir) and `D:` (repository checkout) drives, causing `SHA1 not calculated` errors when running Expo CLI from source. I've tested this in: https://github.com/expo/expo/pull/33204/commits/cbe4ca09ccccc28131b8c2a892813394be8bbc91 - which finally got [Windows to pass all E2E tests](https://github.com/expo/expo/actions/runs/12168114455/job/33938213550)

Reply:
Both done!